src/assets/js/irma-functions.js

```diff
@@ -48,6 +50,29 @@ function irmaIssueCredential(credential, attributes, header = 'Issuing credentia
   });
 }
 
+function irmaIssueRevocableCredential(credential, attributes, revocationKey, header = 'Issuing credential with', disclosePayload = null) {
+  console.log('issueCredential');
+  let validity = {};
+  if (attributes.validity) {
+    validity = {validity: attributes.validity};
+    delete attributes.validity;
+  }
+  const request = {
+    '@context': 'https://irma.app/ld/request/issuance/v2',
+    'credentials': [{
+      credential: credential,
+      attributes: attributes,
+      revocationKey: revocationKey,
+      ...validity,
+    }],
+    'disclose': disclosePayload,
+  };
```

Comment:
This is a part we can extract from both functions into a third function to create the request based on the arguments.

Reply:
Could we also fold it back into a single function, giving it the same treatment as validity? I think that might actually be tidier than two functions, if `revocationKey` can be passed in `attributes`

Reply:
Yeah, we can do that.